jbake-core/src/main/java/org/jbake/parser/MarkdownEngine.java

```diff
@@ -19,53 +35,290 @@
+        Map<String, String> mdOptions = context.getConfig().getMarkdownOptions();
 
         int extensions = PegdownExtensions.NONE;
-
+        List<Extension> flexMakkExtensions = new ArrayList<Extension>();
+        String cleanExtensionName = null;
+        boolean addExt = true;
+        
+        logger.info("Managing custom Markdown Extensions");
+        
         for (String ext : mdExts) {
-            if (ext.startsWith("-")) {
-                ext = ext.substring(1);
-                extensions = removeExtension(extensions, extensionFor(ext));
-            } else {
-                if (ext.startsWith("+")) {
-                    ext = ext.substring(1);
-                }
-                extensions = addExtension(extensions, extensionFor(ext));
+        	if (ext.startsWith("-")) {
+        		cleanExtensionName = ext.substring(1);
+        		addExt = false;
+        	} else if (ext.startsWith("+")) {
+        		cleanExtensionName = ext.substring(1);
+            }else {
+            	cleanExtensionName = ext;
             }
+        	
+        	int pegDownExtension = extensionFor(cleanExtensionName);
+        	
+        	if(pegDownExtension != PegdownExtensions.NONE) {
+        		if (addExt) {
+        			extensions = addExtension(extensions, pegDownExtension);
+                } else {
+                	 extensions = removeExtension(extensions, pegDownExtension);
+                }
+        	}else {
+        		Extension flexMarkExt = searchFlexMarkExtension(cleanExtensionName);
+        		//store it, will be added *after* pegDown options added by adapter
+        		if(null != flexMarkExt) {
+        			flexMakkExtensions.add(flexMarkExt);
+        		}else {
+        			logger.warn("MarkDown extension '{}', NOT found", cleanExtensionName);
+        		}
+        	}
         }
-
-        DataHolder options = PegdownOptionsAdapter.flexmarkOptions(extensions);
-
-        Parser parser = Parser.builder(options).build();
-        HtmlRenderer renderer = HtmlRenderer.builder(options).build();
+        
+        MutableDataSet allOptions = new MutableDataSet();
+        
+        DataHolder pegDownOptions = PegdownOptionsAdapter.flexmarkOptions(extensions);
+        allOptions.setAll(pegDownOptions);
+        
+        Parser.addExtensions(allOptions, flexMakkExtensions.toArray(new Extension[0]));
+        
+        logger.info("Managing custom Markdown Options");
+        //handle options
+        if (null != mdOptions && ! mdOptions.isEmpty()) {
+			for (Map.Entry<String, String> entry : mdOptions.entrySet()) {
+				String[] attributeDetails = entry.getKey().split("\\.");
+				String classOwner;
+				String fieldName;
+				if(attributeDetails.length == 2) {
+					classOwner = attributeDetails[0];
+					fieldName = attributeDetails[1];
+				}else {
+					logger.info("MarkDown option '{}', dosen't have a owner ALL holder with that attribute will be changed and may cause strange beahviors", attributeDetails.toString());
```

Comment:
Please reformat this file .... what is the info log supposed to tell the user?
"Markdown option '' does not have an ALL owner. You may experience strange behaviour as this attribute is applied in a different manner"?